### PR TITLE
replace `package_available` with `module_available`

### LINKED ISSUE #16810: ModuleNotFoundError: Neither `tensorboard` nor `tensorboardX` is available.
### Bug description

`TensorBoardLogger` throws `TensorBoard` not fond error, even though `Tensorboard` is installed.

### How to reproduce the bug

```python
import tensorboard
import torch
from torch import nn
from torch.nn import functional as F
from torch.utils.data import DataLoader
from torch.utils.data import random_split
from torchvision.datasets import MNIST
from torchvision import transforms
import pytorch_lightning as pl
class LitAutoEncoder(pl.LightningModule):
	def __init__(self):
		super().__init__()
		self.encoder = nn.Sequential(
      nn.Linear(28 * 28, 64),
      nn.ReLU(),
      nn.Linear(64, 3))
		self.decoder = nn.Sequential(
      nn.Linear(3, 64),
      nn.ReLU(),
      nn.Linear(64, 28 * 28))

	def forward(self, x):
		embedding = self.encoder(x)
		return embedding

	def configure_optimizers(self):
		optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
		return optimizer

	def training_step(self, train_batch, batch_idx):
		x, y = train_batch
		x = x.view(x.size(0), -1)
		z = self.encoder(x)    
		x_hat = self.decoder(z)
		loss = F.mse_loss(x_hat, x)
		self.log('train_loss', loss)
		return loss

	def validation_step(self, val_batch, batch_idx):
		x, y = val_batch
		x = x.view(x.size(0), -1)
		z = self.encoder(x)
		x_hat = self.decoder(z)
		loss = F.mse_loss(x_hat, x)
		self.log('val_loss', loss)

# data
dataset = MNIST('', train=True, download=True, transform=transforms.ToTensor())
mnist_train, mnist_val = random_split(dataset, [55000, 5000])

train_loader = DataLoader(mnist_train, batch_size=32)
val_loader = DataLoader(mnist_val, batch_size=32)

# model
model = LitAutoEncoder()

# training
logger = pl.loggers.tensorboard.TensorBoardLogger("tb_logs", "model")
trainer = pl.Trainer(precision=16, limit_train_batches=0.5, logger=logger)
trainer.fit(model, train_loader, val_loader)
```

### Error messages and logs

```
    logger = pl.loggers.tensorboard.TensorBoardLogger("tb_logs", "model")
  File "E:\Users\Me\miniconda3\lib\site-packages\pytorch_lightning\loggers\tensorboard.py", line 110, in __init__
    super().__init__(
  File "E:\Users\Me\miniconda3\lib\site-packages\lightning_fabric\loggers\tensorboard.py", line 93, in __init__  
    raise ModuleNotFoundError(
ModuleNotFoundError: Neither `tensorboard` nor `tensorboardX` is available. Try `pip install`ing either.
```

### Environment

<details>
  <summary>Current environment</summary>

```
* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 3060 Ti
        - available:         True
        - version:           11.7
* Lightning:
        - lightning-utilities: 0.6.0.post0
        - pytorch-lightning: 1.9.2
        - torch:             1.13.1
        - torchaudio:        0.13.1
        - torchmetrics:      0.11.1
        - torchvision:       0.14.1
* Packages:
        - absl-py:           1.3.0
        - aiohttp:           3.8.3
        - aiosignal:         1.2.0
        - appdirs:           1.4.4
        - astunparse:        1.6.3
        - async-timeout:     4.0.2
        - attrs:             22.1.0
        - blinker:           1.4
        - bottleneck:        1.3.5
        - brotlipy:          0.7.0
        - cachetools:        4.2.2
        - certifi:           2022.12.7
        - cffi:              1.15.1
        - charset-normalizer: 2.0.4
        - click:             8.0.4
        - colorama:          0.4.6
        - conda:             23.1.0
        - conda-content-trust: 0.1.3
        - conda-package-handling: 2.0.2
        - conda-package-streaming: 0.7.0
        - contourpy:         1.0.5
        - cryptography:      38.0.4
        - cycler:            0.11.0
        - flatbuffers:       2.0
        - flit-core:         3.6.0
        - fonttools:         4.25.0
        - frozenlist:        1.3.3
        - fsspec:            2023.1.0
        - gast:              0.4.0
        - google-auth:       2.6.0
        - google-auth-oauthlib: 0.4.4
        - google-pasta:      0.2.0
        - grpcio:            1.42.0
        - h5py:              3.7.0
        - idna:              3.4
        - keras:             2.10.0
        - keras-preprocessing: 1.1.2
        - kiwisolver:        1.4.4
        - lightning-utilities: 0.6.0.post0
        - markdown:          3.4.1
        - markupsafe:        2.1.1
        - matplotlib:        3.6.2
        - menuinst:          1.4.19
        - mkl-fft:           1.3.1
        - mkl-random:        1.2.2
        - mkl-service:       2.4.0
        - multidict:         6.0.2
        - munkres:           1.1.4
        - numexpr:           2.8.4
        - numpy:             1.23.5
        - oauthlib:          3.2.1
        - opt-einsum:        3.3.0
        - packaging:         22.0
        - pandas:            1.5.2
        - pillow:            9.3.0
        - pip:               22.3.1
        - pluggy:            1.0.0
        - ply:               3.11
        - pooch:             1.4.0
        - protobuf:          3.20.3
        - pyasn1:            0.4.8
        - pyasn1-modules:    0.2.8
        - pycosat:           0.6.4
        - pycparser:         2.21
        - pyjwt:             2.4.0
        - pyopenssl:         22.0.0
        - pyparsing:         3.0.9
        - pyqt5:             5.15.7
        - pyqt5-sip:         12.11.0
        - pysocks:           1.7.1
        - python-dateutil:   2.8.2
        - pytorch-lightning: 1.9.2
        - pytz:              2022.7
        - pyyaml:            6.0
        - requests:          2.28.1
        - requests-oauthlib: 1.3.0
        - rsa:               4.7.2
        - ruamel.yaml:       0.17.21
        - ruamel.yaml.clib:  0.2.6
        - scipy:             1.10.0
        - setuptools:        65.6.3
        - sip:               6.6.2
        - six:               1.16.0
        - tensorboard:       2.10.0
        - tensorboard-data-server: 0.6.1
        - tensorboard-plugin-wit: 1.8.1
        - tensorflow:        2.10.0
        - tensorflow-estimator: 2.10.0
        - termcolor:         2.1.0
        - toml:              0.10.2
        - toolz:             0.12.0
        - torch:             1.13.1
        - torchaudio:        0.13.1
        - torchmetrics:      0.11.1
        - torchvision:       0.14.1
        - tornado:           6.2
        - tqdm:              4.64.1
        - typing-extensions: 4.4.0
        - urllib3:           1.26.14
        - werkzeug:          2.2.2
        - wheel:             0.37.1
        - win-inet-pton:     1.1.0
        - wincertstore:      0.2
        - wrapt:             1.14.1
        - yarl:              1.8.1
        - zstandard:         0.18.0
* System:
        - OS:                Windows
        - architecture:
                - 64bit
                - WindowsPE
        - processor:         AMD64 Family 23 Model 17 Stepping 0, AuthenticAMD
        - python:            3.10.9
        - version:           10.0.18363
```

</details>

### More info

_No response_

cc @tchaton @awaelchli

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

switch to rather use `module_available` until this is resolved: https://github.com/Lightning-AI/lightning/pull/16595#issuecomment-1413347875
resolves #16810

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

**Comment by Borda** [comment]
> Have you checked why the spec is None for the lightning package? See the implementation:

yes, will check just want to patch it until we have it resolved :)

File: src/lightning/pytorch/utilities/imports.py [hunk 1]
```diff
@@ -16,20 +16,20 @@
...
 
-_HABANA_FRAMEWORK_AVAILABLE = package_available("habana_frameworks")
+_HABANA_FRAMEWORK_AVAILABLE = module_available("habana_frameworks")
 _KINETO_AVAILABLE = torch.profiler.kineto_available()
-_OMEGACONF_AVAILABLE = package_available("omegaconf")
-_POPTORCH_AVAILABLE = package_available("poptorch")
-_PSUTIL_AVAILABLE = package_available("psutil")
-_RICH_AVAILABLE = package_available("rich") and compare_version("rich", operator.ge, "10.2.2")
+_OMEGACONF_AVAILABLE = module_available("omegaconf")
+_POPTORCH_AVAILABLE = module_available("poptorch")
```
Comment (carmocca) [inline src/lightning/pytorch/utilities/imports.py:30]: Why update these? Isnt it just a problem with "lightning"?

Comment (Borda) [reply src/lightning/pytorch/utilities/imports.py:30]: I am not sure, so I would rather switch all :otter:

Comment (carmocca) [reply src/lightning/pytorch/utilities/imports.py:30]: We should keep what's working. This must be an issue with `lightning`, not `package_available` as Adrian also thought

File: src/lightning/pytorch/loggers/neptune.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 from lightning.pytorch.utilities.model_summary import ModelSummary
 from lightning.pytorch.utilities.rank_zero import rank_zero_only
 
-_NEPTUNE_AVAILABLE = RequirementCache("neptune-client")
+_NEPTUNE_AVAILABLE = ModuleAvailableCache("neptune-client")
```
Comment (awaelchli) [inline src/lightning/pytorch/loggers/neptune.py:37]: this is not a module

File: tests/tests_pytorch/models/test_hparams.py [hunk 3]
```diff
@@ -691,7 +691,7 @@ def test_model_with_fsspec_as_parameter(tmpdir):
     trainer.test()
 
 
-@pytest.mark.skipif(RequirementCache("hydra-core<1.1"), reason="Requires Hydra's Compose API")
+@pytest.mark.skipif(ModuleAvailableCache("hydra-core<1.1"), reason="Requires Hydra's Compose API")
```
Comment (awaelchli) [inline tests/tests_pytorch/models/test_hparams.py:694]: hydra-core is not a module

**Review by awaelchli — CHANGES_REQUESTED** [review]
There are incorrect usages of the module_available utilities.

**Pull request closed without merging**